src/Test.FunctionalTests.Sarif/TestData/Partitioning/FilterByPredicate/Inputs/FilterByPredicateWithArtifacts.sarif

```diff
@@ -0,0 +1,113 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.4.json",
+  "version": "2.1.0",
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "Sarif Functional Testing"
+        }
+      },
+      "artifacts": [
+        {
+          "location": {
+            "uri": "file1"
+          }
+        },
+        {
+          "location": {
+            "uri": "file2"
+          }
+        },
+        {
+          "location": {
+            "uri": "file3"
+          }
+        },
+        {
+          "location": {
+            "uri": "file4"
+          }
+        }
+      ],
+      "results": [
+        {
+          "ruleId": "TST0001",
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "index": 0
+                }
+              }
+            },
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "index": 2
+                }
+              }
+            }
+          ],
+          "message": {
+            "text": "Message 1.1."
+          }
+        },
+        {
+          "ruleId": "TST0002",
+          "message": {
+            "text": "Message 1.2."
+          }
+        }
+      ],
+      "columnKind": "utf16CodeUnits"
+    },
+    {
+      "tool": {
+        "driver": {
+          "name": "Sarif Functional Testing"
+        }
+      },
+      "artifacts": [
+        {
+          "location": {
+            "uri": "file1"
+          }
+        },
+        {
+          "location": {
+            "uri": "file2"
+          }
+        },
+        {
+          "location": {
+            "uri": "file3"
+          }
+        },
+        {
+          "location": {
+            "uri": "file4"
+          }
+        }
+      ],
+      "results": [
+        {
+          "ruleId": "TST0001",
+          "message": {
+            "text": "Message 2.1."
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "index": 3
+                }
+              }
+            }
+          ]
+        }
+      ],
+      "columnKind": "utf16CodeUnits"
```

Comment:
Maybe include a second Result which uses the same index, but will not match the predicate.
Include a third Result with the same index which will be included, so you see it doesn't get a second copy of the Artifact added.
Include a fourth Result which will be included and has the Artifact Uri inline (and doesn't use ArtifactLocation.Index).

Reply:
All good test cases, done.

---
In reply to: [338202341](https://github.com/microsoft/sarif-sdk/pull/1708#discussion_r338202341) [](ancestors = 338202341)